src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp

```diff
@@ -97,36 +97,85 @@ double get_temporal_id_value(const T& id) {
 CREATE_IS_CALLABLE(apply)
 CREATE_IS_CALLABLE_V(apply)
 
-// apply_callback accomplishes the overload for the
-// two signatures of callback functions.
+template <typename Callbacks, typename Box, typename Cache, typename TemporalId>
+struct AllSimpleCallbacks;
+
+template <typename... Callbacks, typename Box, typename Cache,
+          typename TemporalId>
+struct AllSimpleCallbacks<tmpl::list<Callbacks...>, Box, Cache, TemporalId>
+    : public std::integral_constant<
+          bool, tmpl2::flat_all_v<is_apply_callable_v<Callbacks, Box, Cache,
+                                                      TemporalId>...>> {};
+
 template <typename T, typename DbTags, typename Metavariables,
           typename TemporalId>
-bool apply_callback(
+bool apply_callbacks(
     const gsl::not_null<db::DataBox<DbTags>*> box,
     const gsl::not_null<Parallel::GlobalCache<Metavariables>*> cache,
     const TemporalId& temporal_id) {
-  if constexpr (is_apply_callable_v<typename T::post_interpolation_callback,
-                                    decltype(*box), decltype(*cache),
-                                    decltype(temporal_id)>) {
-    T::post_interpolation_callback::apply(*box, *cache, temporal_id);
+  using callbacks = typename T::post_interpolation_callbacks;
+  constexpr bool all_simple_callbacks =
+      AllSimpleCallbacks<callbacks, decltype(*box), decltype(*cache),
+                         decltype(temporal_id)>::value;
+  if constexpr (all_simple_callbacks) {
+    tmpl::for_each<callbacks>([&](const auto callback_v) {
+      using callback = tmpl::type_from<decltype(callback_v)>;
+      callback::apply(*box, *cache, temporal_id);
+    });
 
     // For the simpler callback function, we will always clean up volume data,
     // so we return true here.
     return true;
   } else {
-    return T::post_interpolation_callback::apply(box, cache, temporal_id);
+    static_assert(tmpl::size<callbacks>::value == 1,
+                  "Can only have 1 post_interpolation_callbacks that mutates "
+                  "the target box.");
+    return tmpl::front<callbacks>::apply(box, cache, temporal_id);
   }
 }
 
 CREATE_HAS_STATIC_MEMBER_VARIABLE(fill_invalid_points_with)
 CREATE_HAS_STATIC_MEMBER_VARIABLE_V(fill_invalid_points_with)
 
+template <typename T, bool B>
+struct with_value;
```

Comment:
This namespace isn't super-specific, so name these structs something more descriptive.